src/configure/auto_outlook.rs

```diff
@@ -1,122 +1,192 @@
-//! Outlook's Autodiscover
+//! # Outlook's Autodiscover
+//!
+//! This module implements autoconfiguration via POX (Plain Old XML) interface to Autodiscover
+//! Service. Newer SOAP interface, introduced in Exchange 2010, is not used.
 
-use quick_xml::events::BytesEnd;
+use quick_xml::events::Event;
+
+use std::io::BufRead;
 
 use crate::context::Context;
-use crate::login_param::LoginParam;
-use crate::provider::Socket;
+use crate::provider::{Protocol, Socket};
 
 use super::read_url::read_url;
-use super::Error;
-
-struct OutlookAutodiscover {
-    pub out: LoginParam,
-    pub out_imap_set: bool,
-    pub out_smtp_set: bool,
-    pub config_type: Option<String>,
-    pub config_server: String,
-    pub config_port: u16,
-    pub config_ssl: String,
-    pub config_redirecturl: Option<String>,
+use super::{Error, ServerParams};
+
+/// Result of parsing a single `Protocol` tag.
+///
+/// https://docs.microsoft.com/en-us/exchange/client-developer/web-service-reference/protocol-pox
+#[derive(Debug)]
+struct ProtocolTag {
+    /// Server type, such as "imap" or "smtp".
+    pub typ: String,
```

Comment:
could this be an enum already here?

Reply:
It may also contain values `WEB`, `POP3`, `EXPR`, `EXCH`, `EXHTTP` etc.

Reply:
I probably need to add a link to https://docs.microsoft.com/en-us/exchange/client-developer/web-service-reference/type-pox and store it as-is, without lowercasing here.

Reply:
👍

Reply:
Changed the code to store strings as-is